standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2433,9 +2426,29 @@ public long allocateTableWriteId(long txnId, String dbName, String tableName) th
   @Override
   public List<TxnToWriteId> allocateTableWriteIdsBatch(List<Long> txnIds, String dbName, String tableName)
           throws TException {
-    AllocateTableWriteIdsRequest rqst = new AllocateTableWriteIdsRequest(txnIds, dbName, tableName);
-    AllocateTableWriteIdsResponse writeIds = client.allocate_table_write_ids(rqst);
-    return writeIds.getTxnToWriteIds();
+    AllocateTableWriteIdsRequest rqst = new AllocateTableWriteIdsRequest(dbName, tableName);
+    rqst.setTxnIds(txnIds);
+    return replAllocateTableWriteIdsBatchIntr(rqst);
+  }
+
+  @Override
+  public List<TxnToWriteId> replAllocateTableWriteIdsBatch(String dbName, String tableName,
+                                           String replPolicy, List<TxnToWriteId> txnToWriteIdList) throws TException {
+    AllocateTableWriteIdsRequest rqst = new AllocateTableWriteIdsRequest(dbName, tableName);
+    rqst.setReplPolicy(replPolicy);
+    rqst.setTxnToWriteIdList(txnToWriteIdList);
+    return replAllocateTableWriteIdsBatchIntr(rqst);
+  }
+
+  private List<TxnToWriteId> replAllocateTableWriteIdsBatchIntr(AllocateTableWriteIdsRequest rqst) throws TException {
```

Comment:
This internal method need not carry the prefix "repl" as it is common for both repl and normal flow.

Reply:
can be done